components/nodemanager-service/tests/project_update_test.go

```diff
@@ -0,0 +1,2231 @@
+package manager
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	iam_v2 "github.com/chef/automate/api/interservice/authz/v2"
+	"github.com/chef/automate/components/nodemanager-service/api/manager"
+	"github.com/chef/automate/components/nodemanager-service/api/nodes"
+	"github.com/chef/automate/components/nodemanager-service/config"
+	"github.com/chef/automate/components/nodemanager-service/pgdb"
+	"github.com/gofrs/uuid"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestProjectUpdate(t *testing.T) {
+	ctx := context.Background()
+	db, err := createPGDB()
+	require.NoError(t, err)
+
+	timestamp, err := ptypes.TimestampProto(time.Now())
+	require.NoError(t, err)
+
+	cases := []struct {
+		description        string
+		projectsData       []*nodes.ProjectsData
+		projectRules       map[string]*iam_v2.ProjectRules
+		originalProjectIDs []string
+		expectedProjectIDs []string
+	}{
+		// Environment
+		{
+			description: "Environment - updating project",
+			projectsData: []*nodes.ProjectsData{
+				{Key: "environment", Values: []string{"env1"}},
+			},
+			projectRules: map[string]*iam_v2.ProjectRules{
+				"project3": {
+					Rules: []*iam_v2.ProjectRule{
+						{
+							Type: iam_v2.ProjectRuleTypes_NODE,
+							Conditions: []*iam_v2.Condition{
+								{
+									Attribute: iam_v2.ProjectRuleConditionAttributes_ENVIRONMENT,
+									Values:    []string{"env1"},
+								},
+							},
+						},
+					},
+				},
+			},
+			originalProjectIDs: []string{"project9"},
+			expectedProjectIDs: []string{"project3"},
+		},
+		{
+			description: "Environment - not updating project. wrong rule type",
+			projectsData: []*nodes.ProjectsData{
+				{Key: "environment", Values: []string{"env1"}},
+			},
+			projectRules: map[string]*iam_v2.ProjectRules{
+				"project3": {
+					Rules: []*iam_v2.ProjectRule{
+						{
+							Type: iam_v2.ProjectRuleTypes_EVENT,
+							Conditions: []*iam_v2.Condition{
+								{
+									Attribute: iam_v2.ProjectRuleConditionAttributes_ENVIRONMENT,
+									Values:    []string{"env1"},
+								},
+							},
+						},
+					},
+				},
+			},
+			originalProjectIDs: []string{"project9"},
+			expectedProjectIDs: []string{},
```

Comment:
😮 If this expectation is correct (getting no projects rather than retaining the original project), then the description is misleading. It does, in fact, update the project to none because of no match.

Reply:
The description is incorrect. I will remove the original project.